v8go.cc

```diff
@@ -93,6 +95,28 @@ RtnError ExceptionError(TryCatch& try_catch, Isolate* iso, Local<Context> ctx) {
   return rtn;
 }
 
+ValuePtr tracked_value(m_ctx* ctx, m_value* val) {
+  // (rogchap) we track values against a context so that when the context is
+  // closed (either manually or GC'd by Go) we can also release all the
+  // values associated with the context; previously the Go GC would not run
+  // quickly enough, as it has no understanding of the C memory allocation size.
+  // By doing so we hold pointers to all values that are created/returned to Go
+  // until the context is released; this is a compromise.
+  // Ideally we would be able to delete the value object and cancel the
+  // finalizer on the Go side, but we currently don't pass the Go ptr, but
+  // rather the C ptr. A potential future iteration would be to use an
+  // unordered_map, where we could do O(1) lookups for the value, but then know
+  // if the object has been finalized or not by being in the map or not. This
+  // would require some ref id for the value rather than passing the ptr between
+  // Go <--> C, which would be a significant change, as there are places where
+  // we get the context from the value, but if we then need the context to get
+  // the value, we would be in a circular bind.
+  ValuePtr val_ptr = static_cast<ValuePtr>(val);
+  ctx->vals.push_back(val_ptr);
+
+  return static_cast<ValuePtr>(val);
```

Comment:
Does this create a new ValuePtr? Is it the same as `val_ptr` above or actually meant to be different?

Reply:
It does not create a new `ValuePtr`; `ValuePtr` is an alias to `void *` aka a pointer. 
So this is just doing the same thing as L114. 
Thanks for picking this up; although not effecting the code, it is a typo and I should just return `val_ptr` for simplicity.